README.md

```diff
@@ -4,15 +4,145 @@
 [![codecov](https://codecov.io/gh/dojo/dgrid/branch/master/graph/badge.svg)](https://codecov.io/gh/dojo/dgrid)
 [![npm version](https://badge.fury.io/js/%40dojo%2Fdgrid.svg)](http://badge.fury.io/js/%40dojo%2Fdgrid)
 
-TODO: Replace with a description of this package
+dgrid is a reactive, unidirectional grid component built using [Dojo 2](https://github.com/dojo/widget-core).
 
 ## Features
 
-TODO: Add sections on features of this package
-
-## How do I use this package?
-
-TODO: Add appropriate usage and instruction guidelines
+- [Lightweight interface](src/providers/README.md) to use any underlying data format or store
+- Customizable columns and cells
+- Sortable columns
+
+![Image of basic grid](http://placekitten.com/450/300)
+
+## Data Providers
+
+Each grid must be passed a `dataProvider`. dgrid ships with a [data provider](src/providers/ArrayDataProvider.ts) to read array data.
+
+## Column Configuration
+
+Each grid must be passed an array of `Column` objects.
+
+`Column<T = any, V = string>` has two optional generics:
+
+- `T`: The data type contained in each row
+- `V`: The data type returned by `get`
+
+Each column supports properties that allow customization:
+
+- `id: string`: Unique to each column, used if `field` is not provided. Useful if a field appears twice
+- `field?: keyof T`: Indicates the property to use to retrieve this column's value from data
+- `get?: V`: Use a static value for this column's value
+- `get?(item: ItemProperties<T>, column: Column<T>): V`: Use a dynamic value for this column's value
+- `label?: string`: Used to display the column's label in the grid header
+- `sortable?: boolean`: Whether this column is sortable. Default: `true`
+- `render?(options: ColumnRenderOptions<T, V>): DNode`: Used to render dynamic content. Passed a value retrieved from data or `get`
```

Comment:
Not clear here where data comes from (at least it's not explicitly explained yet)

Reply:
Perhaps it is clearer later on